### DISCUSSION TITLE
License - Modified Apache 2.0 vs Apache 2.0

### DISCUSSION THREAD

**Discussion by lreading** [maxGraph/maxGraph#35] [Development]
It looks like section 4(e) was [added to the license in October 29, 2020](https://github.com/jgraph/mxgraph/commit/937537c2128daca18dd532cefae57a37d814ba71#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7), which is a clause on work and derivative works being integrated or used with Atlassian products.

The commit includes:
> Note that all versions prior to this commit still fall under the Apache 2.0 license only.

I'm not a lawyer, but I _believe_ the 4(e) clause would prohibit anything using mxgraph from having any kind of Atlassian integration, for example a connector to create Jira tickets.  

If all versions prior to that commit still fall under the Apache 2.0 proper license, would we be able to rebase this fork off of the commit prior to this?  I'm sure there's been a bit of work that's gone into the source since that commit, which would require a bit of effort to get those individual changes back. I think the benefit of a less restrictive license license would be worth it in the long-term.

I can't stress enough that I'm not a lawyer and this isn't legal advice. :sweat_smile:

**Comment by tbouffard** [comment]
I agree with that, as I mentioned in https://github.com/jsGraph/mxgraph/pull/33#issuecomment-833409510, the latest 4.2.2 version is using a regular Apache-2.0 without extra clause.
Commits added after the the license changes in the main branch are mostly taken from code proposal in issues (most of them originally opened in the former development repository jgraph/mxgraph2 no more publicly available), so I guess we could reapply them manually. That being said, as the whole codebase in under refactoring, we may not have this issue on the `development` branch.

About a rebase, IHMO that would break all forks and locally cloned repository, will cause issue when merging the `development` branch into master (at least the first time, maybe we could manage this).
Why not just reverting the commits that fall under the modified license?

Applied commit after the license change
- https://github.com/jgraph/mxgraph/commit/c85074e6b51cd9e2ca33327c1168ff9acf3b1f39: fix for https://github.com/jgraph/mxgraph/issues/507
- https://github.com/jgraph/mxgraph/commit/82321b3496bee84846e47e999619f8a6a374d810: fix for https://github.com/jgraph/mxgraph/issues/510
- https://github.com/jgraph/mxgraph/commit/a5ac1ab1753310426c65f66229e62cab83aa963a: fix for https://github.com/jgraph/mxgraph/issues/506
- https://github.com/jgraph/mxgraph/commit/c85074e6b51cd9e2ca33327c1168ff9acf3b1f39: fix for https://github.com/jgraph/mxgraph/issues/462

**Reply by lreading** [reply]
Rebase may not be the best option, agreed. I didn't see that issue when I had first looked before opening the discussion, my apologies! I think reverting the commits would definitely be a great way of going about it.

**Reply by jgadsden** [reply]
I had an email exchange with @coclav on this and it really is just the name 'mxgraph' that is the problem. 

"
_So I had the confirmation from the guys at mxGraph that we cannot use the name mxGraph, appart from that we are all good. I posted a message about it there : https://github.com/jsGraph/mxgraph/discussions/23_ 
"

so long as we change the name, then all is good and we can use this code base with a new name. Something to do with : 

"//SEIBERT/MEDIA GmbH, Wiesbaden, Germany is the exclusive licensee of 
JGraph for software products based on this codebase within the Atlassian 
ecosystem of products."

but we have been assured that a simple name change will satisfy everyone

**Reply by jgadsden** [reply]
I have to say that it is beyond my understanding ... but a lot of things are

**Reply by Kurt29** [reply]
That seems so wrong and feels so weird

**Comment by lalicw** [comment]
so we can rollback to the original apache license v2?

**Reply by tbouffard** [reply]
Sorry for the late answer
Yes we can rollback to the original apache license v2 it in the `development` branch.
@lalicw do you want to provide a Pull Request?

**Reply by tbouffard** [reply]
done in #85

**Comment by aminosbh** [comment]
I found this discussion too late and I fill like re-doing things already done ! In all cases I created an issue #89 describing a possible process for the transition to the Apache 2.0 license and made #90 and #91 in the same context. I noticed that @tbouffard stated https://github.com/maxGraph/maxGraph/discussions/35#discussioncomment-720142 that the 4 problematic commits could be maybe re-written or re-applied 
> mostly taken from code proposal in issues

**Comment by tbouffard** [comment]
Conversation locked, discussions continue in #89
